Feat: Initial implementation

ModeSevenIndustrialSolutions opened the pull request:
Tested here: https://github.com/modeseven-lfreleng-actions/http-api-verify-action-dev/actions/runs/15539561439

ModeSevenIndustrialSolutions commented:
Also tested downstream here:
https://github.com/lfreleng-actions/chartmuseum-action/pull/4 (draft downstream change refers to this initial-update branch)
https://github.com/lfreleng-actions/chartmuseum-action/actions/runs/15208459781/job/42777429782

tykeal commented on action.yaml:
```diff
@@ -2,37 +2,257 @@
+        if ! [[ "${{ inputs.retries }}" =~ ^[0-9]+$ ]]; then
+          echo "Error: number of retries must be a positive integer ❌"; exit 1
+        fi
+        if ! [[ "${{ inputs.http_code }}" =~ ^[0-9]+$ ]]; then
```
This should be `^[1-5][0-9]{2}$` as all HTTP response codes are 3 digits and are 1xx-5xx

tykeal commented on action.yaml:
```diff
@@ -2,37 +2,257 @@
+          echo "Error: number of retries must be a positive integer ❌"; exit 1
+        fi
+        if ! [[ "${{ inputs.http_code }}" =~ ^[0-9]+$ ]]; then
+          echo "Error: HTTP response code must be a positive integer ❌"; exit 1
```
`Error: HTTP response code must be a proper status class 1xx-5xx`

ModeSevenIndustrialSolutions replied:
Think this is an accidental duplicate of the above line.

tykeal commented on action.yaml:
```diff
@@ -2,37 +2,257 @@
+    # With default one second retries, will timeout after 90 seconds
+    type: 'string'
+    default: 60
+  http_code:
```
This should be called `http_response_code` or `expected_http_response_code` as having `response_code` in the name is more clear as to the intended use even without the description text.

tykeal commented on .github/workflows/build-test-release.yaml:
```diff
@@ -0,0 +1,247 @@
+    runs-on: 'ubuntu-24.04'
+    permissions:
+      contents: write
+      # id-token: write
```
If this isn't needed, then please remove the comment version of it as well.

ModeSevenIndustrialSolutions replied:
Removed commented id-token line

tykeal commented on .github/workflows/build-test-release.yaml:
```diff
@@ -0,0 +1,247 @@
+      - 'pypi'
+    runs-on: 'ubuntu-24.04'
+    permissions:
+      contents: write
```
Why is `write` needed?

ModeSevenIndustrialSolutions replied:
Added comment explaining why

tykeal commented on .github/workflows/build-test-release.yaml:
```diff
@@ -0,0 +1,247 @@
+      - 'python-build'
+      - 'promote-release'
+    permissions:
+      contents: write
```
Why is `write` needed?

ModeSevenIndustrialSolutions replied:
Added comment explaining why

tykeal commented on .github/workflows/build-test.yaml:
```diff
@@ -0,0 +1,102 @@
+      artefact_name: "${{ steps.python-build.outputs.artefact_name }}"
+      artefact_path: "${{ steps.python-build.outputs.artefact_path }}"
+    permissions:
+      contents: write
```
Why is `write` needed?

ModeSevenIndustrialSolutions replied:
Build process only needs read permissions; changed in code

ModeSevenIndustrialSolutions committed:
feat: Complete HTTP API tool implementation with Docker support and comprehensive CI/CD

- Implement Python HTTP API testing tool with CLI interface
- Add Docker containerization for easy deployment
- Setup comprehensive GitHub Actions CI/CD pipeline
- Add integration tests with go-httpbin for realistic testing
- Implement SSL/TLS support with custom CA bundles
- Add extensive error handling and retry mechanisms
- Include multiple output formats and debugging capabilities
- Setup proper project structure with PDM dependency management
- Add security scanning and dependency management workflows
- Implement proper caching strategies for builds and tests
